prescriptions/pi_/pillow/pillow3.8.0_new_array.yaml

```diff
@@ -0,0 +1,26 @@
+units:
+  sieves:
+    - name: Pillow380NewArraySieve
+      type: sieve
+      should_include:
+        adviser_pipeline: true
+        dependency_monkey_pipeline: true
+        recommendation_types:
+          - performance
+          - security
+          - stable
+          - latest
+          - test
+      match:
+        - package_version:
+            name: pillow
+            version: "3.8.0"
+            index_url: "https://pypi.org/simple"
+      run:
+        log:
+          message: "Pillow in version 3.8.0 introduced a new array that did not match all numpy array parameters."
+          type: WARNING
+        stack_info:
+          - type: WARNING
+            message: "Pillow in version 3.8.0 filtered out"
```

Comment:
Do we want to filter out `pillow==3.8.0` for all stacks or only for those that are running with specific numpy versions?

Reply:
Should we run DM with a new script, to discover which versions are affected by the error `TypeError: __array__() takes 1 positional argument but 2 were given`? It seems the issue was introduced by Pillow, the logic in Numpy was there, but don't since when.

Reply:
Any and all numpy versions.

Reply:
thank you very much @t-vi!

Reply:
As the issue is dependent on `numpy` presence, this should be turned into a step pipeline unit and corresponding pillow version should be filtered only based on numpy presence.

https://thoth-station.ninja/docs/developers/adviser/prescription.html#steps

Otherwise, resolver will filter out all the pillow==3.8.0 versions, even for stacks that do not have the issue stated.